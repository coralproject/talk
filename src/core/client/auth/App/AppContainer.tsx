import React, { Component } from "react";

import {
  graphql,
  withFragmentContainer,
  withLocalStateContainer,
} from "coral-framework/lib/relay";

import { AppContainer_auth as AuthData } from "coral-auth/__generated__/AppContainer_auth.graphql";
import { AppContainer_viewer as UserData } from "coral-auth/__generated__/AppContainer_viewer.graphql";
import { AppContainerLocal as Local } from "coral-auth/__generated__/AppContainerLocal.graphql";

import AccountCompletionContainer from "./AccountCompletion";
import App from "./App";

interface Props {
  local: Local;
  auth: AuthData;
  viewer: UserData | null;
}

class AppContainer extends Component<Props> {
  public render() {
    const {
      local: { view },
      auth,
      viewer,
    } = this.props;

    // If we're dealing with a password reset, we can't possibly worry about
    // account completion (because they are not logged in, or have already
    // completed their account), so disregard here, and just return the App.
    if (view === "FORGOT_PASSWORD") {
      return <App view={view} auth={auth} viewer={viewer} />;
    }

    return (
      <AccountCompletionContainer auth={auth} viewer={viewer}>
        <App view={view} auth={auth} viewer={viewer} />
      </AccountCompletionContainer>
    );
  }
}

const enhanced = withLocalStateContainer(
  graphql`
    fragment AppContainerLocal on Local {
      view
    }
  `
)(
  withFragmentContainer<Props>({
    auth: graphql`
      fragment AppContainer_auth on Auth {
        ...SignInContainer_auth
        ...SignUpContainer_auth
        ...AccountCompletionContainer_auth
      }
    `,
    viewer: graphql`
      fragment AppContainer_viewer on User {
        ...AccountCompletionContainer_viewer
        ...ForgotPasswordContainer_viewer
      }
    `,
  })(AppContainer)
);

export default enhanced;
