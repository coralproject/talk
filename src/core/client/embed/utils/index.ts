export { default as buildURL } from "./buildURL";
export { default as ensureEndSlash } from "./ensureEndSlash";
export { default as ensureNoEndSlash } from "./ensureNoEndSlash";
export { default as startsWith } from "./startsWith";
export { default as prefixStorage } from "./prefixStorage";
export { default as parseHashQuery } from "./parseHashQuery";
