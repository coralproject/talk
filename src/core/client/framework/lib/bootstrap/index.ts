export * from "./TalkContext";
export { default as createContext } from "./createContext";
export { default as withContext } from "./withContext";
