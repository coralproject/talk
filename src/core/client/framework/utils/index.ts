export { default as buildURL } from "./buildURL";
export { default as parseURL } from "./parseURL";
export { default as modifyQuery } from "./modifyQuery";
