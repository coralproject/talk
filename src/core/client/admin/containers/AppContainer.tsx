import { BrowserProtocol, queryMiddleware } from "farce";
import { createFarceRouter, createRender } from "found";
import { Resolver } from "found-relay";

import React, { StatelessComponent } from "react";
import { TalkContextConsumer } from "talk-framework/lib/bootstrap/TalkContext";
import routeConfig from "../routeConfig";
import NotFound from "../routes/NotFound";

const Router = createFarceRouter({
  historyProtocol: new BrowserProtocol(),
  historyMiddlewares: [queryMiddleware],
  routeConfig,
  render: createRender({
    renderError: ({ error }) => (
      <div>{error.status === 404 ? <NotFound /> : "Error"}</div>
    ),
  }),
});

const AppContainer: StatelessComponent = () => (
  <TalkContextConsumer>
    {({ relayEnvironment }) => (
      <Router resolver={new Resolver(relayEnvironment)} />
    )}
  </TalkContextConsumer>
);

export default AppContainer;
