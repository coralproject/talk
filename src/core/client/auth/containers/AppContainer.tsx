import * as React from "react";
import { StatelessComponent } from "react";

import { AppContainerLocal as Local } from "talk-auth/__generated__/AppContainerLocal.graphql";
import { graphql, withLocalStateContainer } from "talk-framework/lib/relay";

import App from "../components/App";

interface InnerProps {
  local: Local;
}

const AppContainer: StatelessComponent<InnerProps> = ({ local: { view } }) => {
  return <App view={view} />;
};

const enhanced = withLocalStateContainer(
  graphql`
    fragment AppContainerLocal on Local {
      view
    }
  `
)(AppContainer);

export default enhanced;
