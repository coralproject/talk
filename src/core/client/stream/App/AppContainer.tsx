import React from "react";

import { graphql, withLocalStateContainer } from "coral-framework/lib/relay";

import { AppContainerLocal as Local } from "coral-stream/__generated__/AppContainerLocal.graphql";

import App from "./App";
import {
  OnEvents,
  OnPostMessageSetAccessToken,
  OnPymLogin,
  OnPymLogout,
  OnPymSetCommentID,
} from "./listeners";

const listeners = (
  <>
    <OnPymLogin />
    <OnPymLogout />
    <OnPymSetCommentID />
    <OnPostMessageSetAccessToken />
    <OnEvents />
  </>
);

interface Props {
  disableListeners?: boolean;
  local: Local;
}

class AppContainer extends React.Component<Props> {
  public render() {
    const {
      local: { activeTab },
    } = this.props;

    return (
      <>
        {this.props.disableListeners ? null : listeners}
        <App activeTab={activeTab} />
      </>
    );
  }
}

const enhanced = withLocalStateContainer(
  graphql`
    fragment AppContainerLocal on Local {
      activeTab
    }
  `
)(AppContainer);

export default enhanced;
