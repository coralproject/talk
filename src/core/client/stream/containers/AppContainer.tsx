import * as React from "react";
import { StatelessComponent } from "react";

import { graphql, withLocalStateContainer } from "talk-framework/lib/relay";
import { AppContainerLocal as Local } from "talk-stream/__generated__/AppContainerLocal.graphql";

import App from "../components/App";

interface InnerProps {
  local: Local;
}

const AppContainer: StatelessComponent<InnerProps> = ({
  local: { commentID },
}) => {
  return <App showPermalinkView={!!commentID} />;
};

const enhanced = withLocalStateContainer<Local>(
  graphql`
    fragment AppContainerLocal on Local {
      commentID
    }
  `
)(AppContainer);

export default enhanced;
