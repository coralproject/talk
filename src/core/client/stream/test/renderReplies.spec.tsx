import React from "react";
import TestRenderer from "react-test-renderer";
import { RecordProxy } from "relay-runtime";
import sinon from "sinon";

import { timeout } from "talk-common/utils";
import { TalkContext, TalkContextProvider } from "talk-framework/lib/bootstrap";
import AppContainer from "talk-stream/containers/AppContainer";

import createEnvironment from "./createEnvironment";
import { assetWithReplies } from "./fixtures";

const resolvers = {
  Query: {
    asset: sinon
      .stub()
      .throws()
      .withArgs(undefined, { id: assetWithReplies.id })
      .returns(assetWithReplies),
  },
};

const environment = createEnvironment({
  // Set this to true, to see graphql responses.
  logNetwork: false,
  resolvers,
  initLocalState: (localRecord: RecordProxy) => {
    localRecord.setValue(assetWithReplies.id, "assetID");
  },
});

const context: TalkContext = {
  relayEnvironment: environment,
  localeMessages: [],
};

const testRenderer = TestRenderer.create(
  <TalkContextProvider value={context}>
    <AppContainer />
  </TalkContextProvider>
);

it("renders comment stream", async () => {
  // Wait for loading.
  await timeout();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
