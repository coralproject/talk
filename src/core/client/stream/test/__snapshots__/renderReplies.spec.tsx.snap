// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders comment stream 1`] = `
<div
  className="Flex-root App-root Flex-justifyCenter Flex-alignCenter"
>
  <div
    className="Stream-root"
  >
    <form
      autoComplete="off"
      onSubmit={[Function]}
    >
      <div>
        <textarea
          className="PostCommentForm-textarea"
          name="body"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="PostCommentForm-postButtonContainer"
      >
        <button
          className="BaseButton-root Button-root Button-sizeRegular Button-colorPrimary Button-variantFilled"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
        >
          Post
        </button>
      </div>
    </form>
    <div
      aria-live="polite"
      className="Flex-root Flex-itemGutter Flex-alignFlexStart Flex-directionColumn"
      id="talk-comments-stream-log"
      role="log"
    >
      <div
        className="Flex-root Flex-itemGutter Flex-alignFlexStart Flex-directionColumn"
      >
        <div
          role="article"
        >
          <div
            className="Flex-root TopBar-root Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn"
          >
            <span
              className="Typography-root Typography-heading3 Typography-colorTextPrimary Username-root"
            >
              Markus
            </span>
            <time
              className="Timestamp-root RelativeTime-root"
              dateTime="2018-07-06T18:24:00.000Z"
              title="2018-07-06T18:24:00.000Z"
            >
              2018-07-06T18:24:00.000Z
            </time>
          </div>
          <p
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            Joining Too
          </p>
          <div
            className="Comment-footer"
          />
        </div>
      </div>
      <div
        className="Flex-root Flex-itemGutter Flex-alignFlexStart Flex-directionColumn"
      >
        <div
          role="article"
        >
          <div
            className="Flex-root TopBar-root Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn"
          >
            <span
              className="Typography-root Typography-heading3 Typography-colorTextPrimary Username-root"
            >
              Markus
            </span>
            <time
              className="Timestamp-root RelativeTime-root"
              dateTime="2018-07-06T18:24:00.000Z"
              title="2018-07-06T18:24:00.000Z"
            >
              2018-07-06T18:24:00.000Z
            </time>
          </div>
          <p
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            I like yoghurt
          </p>
          <div
            className="Comment-footer"
          />
        </div>
        <div
          className="Indent-root Indent-level0"
        >
          <div
            className="Flex-root Flex-itemGutter Flex-alignFlexStart Flex-directionColumn"
            id="talk-comments-replyList-log--comment-with-replies"
            role="log"
          >
            <div
              role="article"
            >
              <div
                className="Flex-root TopBar-root Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn"
              >
                <span
                  className="Typography-root Typography-heading3 Typography-colorTextPrimary Username-root"
                >
                  Markus
                </span>
                <time
                  className="Timestamp-root RelativeTime-root"
                  dateTime="2018-07-06T18:24:00.000Z"
                  title="2018-07-06T18:24:00.000Z"
                >
                  2018-07-06T18:24:00.000Z
                </time>
              </div>
              <p
                className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
              >
                Joining Too
              </p>
              <div
                className="Comment-footer"
              />
            </div>
            <div
              role="article"
            >
              <div
                className="Flex-root TopBar-root Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn"
              >
                <span
                  className="Typography-root Typography-heading3 Typography-colorTextPrimary Username-root"
                >
                  Lukas
                </span>
                <time
                  className="Timestamp-root RelativeTime-root"
                  dateTime="2018-07-06T18:20:00.000Z"
                  title="2018-07-06T18:20:00.000Z"
                >
                  2018-07-06T18:20:00.000Z
                </time>
              </div>
              <p
                className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
              >
                What's up?
              </p>
              <div
                className="Comment-footer"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
